worker/codalabworker/local_run/local_run_manager.py

```diff
@@ -360,3 +360,32 @@ def memory_bytes(self):
         except ValueError:
             # Fallback to sysctl when os.sysconf('SC_PHYS_PAGES') fails on OS X
             return int(check_output(['sysctl', '-n', 'hw.memsize']).strip())
+
+    @property
+    def free_disk_bytes(self):
+        """
+        Available disk space by bytes of this RunManager.
+        """
+        error_msg = "Failed to run command {}".format("df " + self._work_dir)
+        try:
+            p = Popen(["df", self._work_dir], stdout=PIPE)
+            output, error = p.communicate()
+            # Return None when there is an error.
+            if error:
+                logger.error(error.strip())
+                return None
+
+            if output:
+                lines = output.decode().split("\n")
+                header = lines[0]
+                if header.find("Available") > -1 and len(lines) > 1:
```

Comment:
Write `!= -1` to be more precise with the semantics and I think writing `>= 2` instead of `> 1` makes it easier to interpret.

Reply:
Actually, reflecting on this, it's probably better not to have these conditions at all and just let any errors fall to the `except` statement to keep this code cleaner.  I separate errors into two categories: (i) normal errors that happen - these should be handled explicitly with checks and appropriate error messages, and (ii) abnormal errors that suggest we have a bug or something is deeply wrong - if this happens, someone needs to go investigate - for that, it's fine to just have the blanket `except Exception as e`.

Reply:
I think you are right. If str.index() isn't able to find the column "Available", it will throw an exception. If return value from df isn't a valid format, when we cast it to integer, it will also throw an exception. Both cases will be caught by the outer try-catch block.